File: sdk_src/Server/MobileIMSDKServer_Open/src/net/x52im/mobileimsdk/server/protocal/s/PLoginInfoResponse.java
/*
 * Copyright (C) 2021  即时通讯网(52im.net) & Jack Jiang.
 * The MobileIMSDK v6.x Project. 
 * All rights reserved.
 * 
 * > Github地址：https://github.com/JackJiang2011/MobileIMSDK
 * > 文档地址：  http://www.52im.net/forum-89-1.html
 * > 技术社区：  http://www.52im.net/
 * > 技术交流群：320837163 (http://www.52im.net/topic-qqgroup.html)
 * > 作者公众号：“【即时通讯技术圈】”，欢迎关注！
 * > 联系作者：  http://www.52im.net/thread-2792-1-1.html
 *  
 * "即时通讯网(52im.net) - 即时通讯开发者社区!" 推荐开源工程。
 * 
 * PLoginInfoResponse.java at 2021-6-29 10:15:36, code by Jack Jiang.
 */
package net.x52im.mobileimsdk.server.protocal.s;

public class PLoginInfoResponse
{
	protected int code = 0;
	
	protected long firstLoginTime = 0;
	public PLoginInfoResponse(int code, long firstLoginTime)
	{
		this.code = code;
		this.firstLoginTime = firstLoginTime;
	}

	public int getCode()
	{
		return code;
	}
	public void setCode(int code)
	{
		this.code = code;
	}

	public long getFirstLoginTime()
	{
		return firstLoginTime;
	}

	public void setFirstLoginTime(long firstLoginTime)
	{
		this.firstLoginTime = firstLoginTime;
	}
}

File: sdk_src/TCP_Client/MobileIMSDK4f_tcp_Open/lib/server/protocol/s/p_login_info_response.dart
class PLoginInfoResponse {
  int _code = 0;
  int _firstLoginTime = 0;

  PLoginInfoResponse(int code, int firstLoginTime) {
    _code = code;
    _firstLoginTime = firstLoginTime;
  }

  //命名式构造方法，也可以是用工厂构造方法
  PLoginInfoResponse.fromJson(Map<String, dynamic> json)
      : _code = json['code'],
        _firstLoginTime = json['firstLoginTime'];

  //如果想写成协议，归档(json和对象互转时，为了使用方便)时，可以继承协议，那么可以用普通方法，而不是构造方法
  Map<String, dynamic> toJson() =>
      {'code': _code, 'firstLoginTime': _firstLoginTime};

  int getCode() {
    return _code;
  }

  void setCode(int code) {
    _code = code;
  }

  getFirstLoginTime() {
    return _firstLoginTime;
  }

  void setFirstLoginTime(int firstLoginTime) {
    _firstLoginTime = firstLoginTime;
  }
}
